src/xor8.rs

```diff
@@ -205,12 +205,17 @@ where
     }
 
     /// Populate with pre-compute collection of 64-bit digests.
-    pub fn populate_keys(&mut self, digests: &[u64]) {
-        if let Some(x) = self.num_keys.as_mut() {
-            *x += digests.len()
+    pub fn populate_keys<'i, I: Iterator<Item = &'i u64>>(&mut self, digests: I) {
```

Comment:
I think it would be better [to ask for an `IntoIterator` trait](https://rust-lang.github.io/api-guidelines/flexibility.html?highlight=intoiterator#functions-minimize-assumptions-about-parameters-by-using-generics-c-generic) here.

Reply:
Good idea:D Thanks!